isaaclab_arena/tasks/sequential_task_base.py

```diff
@@ -0,0 +1,202 @@
+# All rights reserved.
+#
+# SPDX-License-Identifier: Apache-2.0
+
+import copy
+import dataclasses
+import torch
+from dataclasses import MISSING
+from typing import Any
+
+from isaaclab.managers import EventTermCfg, TerminationTermCfg
+from isaaclab.utils import configclass
+
+from isaaclab_arena.tasks.task_base import TaskBase
+from isaaclab_arena.utils.configclass import combine_configclass_instances, make_configclass
+
+
+@configclass
+class SequentialTaskEventsCfg:
+    reset_subtask_success_state: EventTermCfg = MISSING
+
+
+@configclass
+class TerminationsCfg:
+    success: TerminationTermCfg = MISSING
+
+
+class SequentialTaskBase(TaskBase):
+    """
+    A base class for tasks composed sequentially from multiple subtasks.
+    The sequential task takes a list of TaskBase instances (subtasks),
+    and automatically collects configs to form a composite task.
+    """
+
+    def __init__(self, subtasks: list[TaskBase], episode_length_s: float | None = None):
+        super().__init__(episode_length_s)
+        assert len(subtasks) > 0, "SequentialTaskBase requires at least one subtask"
+        self.subtasks = subtasks
+
+    @staticmethod
+    def _add_suffix_to_configclass_fields(cfg_instance: Any, suffix: str) -> Any:
+        """Create a new configclass instance with all field names appended with a suffix."""
+        if cfg_instance is None:
+            return None
+
+        fields = dataclasses.fields(cfg_instance)
+        new_fields = []
+        field_values = {}
+
+        # Rename the fields with suffix
+        for field in fields:
+            new_name = f"{field.name}{suffix}"
+            value = getattr(cfg_instance, field.name)
+            new_fields.append((new_name, field.type, value))
+            field_values[new_name] = value
+
+        # Create a new configclass with renamed fields
+        new_cfg_class = make_configclass(type(cfg_instance).__name__, new_fields)
+        return new_cfg_class(**field_values)
+
+    @staticmethod
+    def _remove_configclass_fields(cfg_instance: Any, exclude_fields: set[str]) -> Any:
+        "Remove all fields from the configclass instance that are in the exclude_fields set."
+        if cfg_instance is None:
+            return None
+
+        fields = dataclasses.fields(cfg_instance)
+        new_fields = []
+        field_values = {}
+
+        # Remove the fields that are in the exclude_fields set
+        for field in fields:
+            if field.name in exclude_fields:
+                continue
+            value = getattr(cfg_instance, field.name)
+            new_fields.append((field.name, field.type, value))
+            field_values[field.name] = value
+
+        if not new_fields:
+            return None
+
+        # Create a new configclass without the excluded fields
+        new_cfg_class = make_configclass(type(cfg_instance).__name__, new_fields)
+        return new_cfg_class(**field_values)
```

Comment:
As above, suggestion to move this to `isaaclab_arena.utils.configclass`.

Additionally, the two functions above have a very similar form. They loop through the fields, build a new config class by performing some mutation on each field. I wonder if we could write them in terms of a function `transform_config_class` which takes a `Callable`, loops through the fields, calling the callable on each and adding the resulting field to a new `configclass` if the return is not None.